### PR TITLE
Star pattern and score cleanup

### PR DESCRIPTION
**Pull request by Arrcival**

### CODE REVIEW

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs [hunk 1]
```diff
@@ -202,7 +193,8 @@ public void CreatePattern()
             switch (patternID)
             {
                 case 1:
-                    PatternWave();
+                    //PatternWave();
+                    PatternStar();
```
Comment (Shawdooow) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs]: revert this pls

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs [hunk 2]
```diff
@@ -48,17 +48,8 @@ protected override void CheckJudgement(bool userTriggered)
                 Judgement.Result = HitResult.Hit;
                 switch (Score)
```
Comment (ColdVolcano) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs]: Switch statement with 1 entry 🤔

File: osu.Game.Rulesets.Vitaru/Scoring/VitaruScoreProcessor.cs [hunk 3]
```diff
@@ -42,11 +42,8 @@ public override void PopulateScore(Score score)
         {
             base.PopulateScore(score);
 
-            score.Statistics[@"300"] = scoreResultCounts.GetOrDefault(VitaruScoreResult.Graze300);
-            score.Statistics[@"100"] = scoreResultCounts.GetOrDefault(VitaruScoreResult.Graze100);
-            score.Statistics[@"50"] = scoreResultCounts.GetOrDefault(VitaruScoreResult.Graze50);
-            score.Statistics[@"10"] = scoreResultCounts.GetOrDefault(VitaruScoreResult.Graze10);
-            score.Statistics[@"x"] = scoreResultCounts.GetOrDefault(VitaruScoreResult.Miss);
+            score.Statistics[@"Dodges"] = scoreResultCounts.GetOrDefault(VitaruScoreResult.Hit300);
+            score.Statistics[@"Hits"] = scoreResultCounts.GetOrDefault(VitaruScoreResult.Miss);
```
Comment (Shawdooow) [inline osu.Game.Rulesets.Vitaru/Scoring/VitaruScoreProcessor.cs]: "Hit" + "Dodge"

File: osu.Game/OsuGameBase.cs [hunk 4]
```diff
@@ -20,6 +20,7 @@
 using osu.Game.Online.API;
 using SQLite.Net;
 using osu.Framework.Graphics.Performance;
+using System.Linq;
```
Comment (Shawdooow) [inline osu.Game/OsuGameBase.cs]: No

File: osu.Game/OsuGameBase.cs [hunk 5]
```diff
@@ -146,6 +147,10 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
+            RulesetDatabase.AllRulesets.Select(info =>
```
Comment (Shawdooow) [inline osu.Game/OsuGameBase.cs]: No

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs [hunk 6]
```diff
@@ -380,5 +371,39 @@ public void PatternCircle()
             }
             ScorePattern = (int)calculateDifficulty(Pattern.PatternBulletWidth, Pattern.PatternSpeed, numberbullets, false, PatternCount, 0, 1);
         }
+        
+        public void PatternStar()
+        {
+            bool strange = false;
+            int numberbulletsbranch = (int)(Pattern.PatternDifficulty * 2) + 1;
+            int numberbullets = numberbulletsbranch * 5;
```
Comment (ColdVolcano) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs]: I have no idea what's the usage of these two

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs [hunk 7]
```diff
@@ -380,5 +371,39 @@ public void PatternCircle()
...
+        public void PatternStar()
+        {
+            bool strange = false;
+            int numberbulletsbranch = (int)(Pattern.PatternDifficulty * 2) + 1;
+            int numberbullets = numberbulletsbranch * 5;
+            int bullettop = (int)(Pattern.PatternDifficulty + 1);
+            float directionModifier = (float)(360 / numberbullets);
+            float angleModifier = 0f;
+            float speedModifier = 0f;
+            directionModifier = MathHelper.DegreesToRadians(directionModifier);
```
Comment (ColdVolcano) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs]: you could just combine this

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs [hunk 8]
```diff
@@ -380,5 +371,39 @@ public void PatternCircle()
...
             ScorePattern = (int)calculateDifficulty(Pattern.PatternBulletWidth, Pattern.PatternSpeed, numberbullets, false, PatternCount, 0, 1);
         }
+        
+        public void PatternStar()
+        {
+            bool strange = false;
+            int numberbulletsbranch = (int)(Pattern.PatternDifficulty * 2) + 1;
+            int numberbullets = numberbulletsbranch * 5;
+            int bullettop = (int)(Pattern.PatternDifficulty + 1);
+            float directionModifier = (float)(360 / numberbullets);
```
Comment (ColdVolcano) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs]: and this.
Also, use 360f / numberbullets

File: osu.Game.Rulesets.Vitaru/Scoring/VitaruScoreProcessor.cs (hunk 3 above)
Comment (Arrcival) [inline osu.Game.Rulesets.Vitaru/Scoring/VitaruScoreProcessor.cs]: no plural?

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs (hunk 7 above)
Comment (Arrcival) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs]: yes

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs (hunk 6 above)
Comment (Arrcival) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs]: number of bullets in a branch ?
total bullets for ScorePattern at the end ?

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs (hunk 7 above)
Comment (ColdVolcano) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs]: I mean combine 381 and 384

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs (hunk 8 above)
Comment (Arrcival) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs]: bullettop is the n° of the bullet which represent the top of the star branch

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs (hunk 7 above)
Comment (Arrcival) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs]: yes

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs (hunk 6 above)
Comment (ColdVolcano) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawablePatterns.cs]: I guess it should be totalBullets then

File: osu.Game/OsuGameBase.cs (hunk 5 above)
Comment (Arrcival) [inline osu.Game/OsuGameBase.cs]: So..sorry Onii-chan ;-;

File: osu.Game/OsuGameBase.cs (hunk 4 above)
Comment (Arrcival) [inline osu.Game/OsuGameBase.cs]: You not like Link?

**Pull request closed without merging**